src/app/index.js

```diff
@@ -27,11 +13,28 @@ var app = angular.module('mainApp', ['ngRoute'])
         // this.favorite_list.push(new Song('Jose Sabía', 'La Vela Puerca', 'De Bichos y Flores'));
         
         this.has_favorites = function(){
-            return this.favorite_list.length > 0;
-        }.bind(this);
+            return Favorite_Serv.favorite_list.length > 0;
```

Comment:
dont use snake case, use camelCase only when dealing with variable and function names! when dealing with files use hyphen instead of snake case. ie: 

`var myVar = true;`
`function myFunction () { }`
`/my-app/controllers/my-controller.js`